src/main/java/org/creekservice/kafka/test/perf/implementations/DevHarrelImplementation.java

```diff
@@ -0,0 +1,146 @@
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Set;
+import org.creekservice.kafka.test.perf.TestSchemas;
+import org.creekservice.kafka.test.perf.model.TestModel;
+import org.creekservice.kafka.test.perf.testsuite.AdditionalSchemas;
+import org.creekservice.kafka.test.perf.testsuite.SchemaSpec;
+
+@SuppressWarnings("FieldMayBeFinal") // not final to avoid folding.
+public class DevHarrelImplementation implements Implementation {
+
+    private static final MetaData METADATA =
+            new MetaData(
+                    "json-schema (dev.harrel)",
+                    "json-schema (dev.harrel)",
+                    Language.Java,
+                    Licence.MIT,
+                    Set.of(DRAFT_2020_12, DRAFT_2019_09),
+                    "https://github.com/harrel56/json-schema");
+
+    private final ObjectMapper mapper = JsonMapper.builder().build();
+    private final Map<SchemaSpec, Validator> validators;
+    private final URI testSchemaUri = URI.create("urn:test");
+    private Map<URI, String> remotes = Map.of();
+
+    public DevHarrelImplementation() {
+        final SchemaResolver schemaResolver =
+                uri -> {
+                    final String resolved = remotes.get(URI.create(uri));
+                    if (resolved == null) {
+                        return SchemaResolver.Result.empty();
+                    }
+                    return SchemaResolver.Result.fromString(resolved);
+                };
+        final Validator validator2020 =
+                new dev.harrel.jsonschema.ValidatorFactory()
+                        .withSchemaResolver(schemaResolver)
+                        .createValidator();
+        final Validator validator2019 =
+                new dev.harrel.jsonschema.ValidatorFactory()
+                        .withDialect(new Dialects.Draft2019Dialect())
+                        .withSchemaResolver(schemaResolver)
+                        .createValidator();
+        /* Validate against meta-schemas in order to parse them eagerly */
+        validator2020.validate(URI.create(SpecificationVersion.DRAFT2020_12.getId()), "{}");
+        validator2019.validate(URI.create(SpecificationVersion.DRAFT2019_09.getId()), "{}");
+
+        validator2020.registerSchema(testSchemaUri, TestSchemas.DRAFT_2020_SCHEMA);
+
+        this.validators =
+                Map.of(
+                        DRAFT_2020_12, validator2020,
+                        DRAFT_2019_09, validator2019);
+    }
+
+    @Override
+    public MetaData metadata() {
+        return METADATA;
+    }
+
+    @Override
+    public JsonValidator prepare(
+            final String schema, final SchemaSpec spec, final AdditionalSchemas additionalSchemas) {
+        DevHarrelImplementation.this.remotes = additionalSchemas.remotes();
+        final Validator validator = validators.get(spec);
+        final URI schemaUri = validator.registerSchema(schema);
```

Comment:
With the constructor gone, you'll need something like:

```suggestion
        final Validator validator = validator(spec, additionalSchemas);       
        final URI schemaUri = validator.registerSchema(schema);
```

With:

```java
private Validator validator(final SchemaSpec spec, final AdditionalSchemas additionalSchemas) {
   final SchemaResolver schemaResolver =
                uri -> {
                    final String resolved = additionalSchemas.remotes().get(URI.create(uri));
                    if (resolved == null) {
                        return SchemaResolver.Result.empty();
                    }
                    return SchemaResolver.Result.fromString(resolved);
                };

    switch (spec) {
      case: SchemaSpec.DRAFT_2020_12:
           final Validator validator = new dev.harrel.jsonschema.ValidatorFactory()
                        .withSchemaResolver(schemaResolver)
                        .createValidator();
            /* Validate against meta-schemas in order to parse them eagerly */
           // Note: If `registerSchema()` will cause this to be parsed, then I don't think this will be necessary, as schema parsing is not currently part of performance benchmark
            validator.validate(URI.create(SpecificationVersion.DRAFT2020_12.getId()), "{}");
            return validator;
      case: SchemaSpec.DRAFT_2019_09:
           final Validator validator = new dev.harrel.jsonschema.ValidatorFactory()
                        .withDialect(new Dialects.Draft2019Dialect())
                        .withSchemaResolver(schemaResolver)
                        .createValidator();
            /* Validate against meta-schemas in order to parse them eagerly */
             // Note: If `registerSchema()` will cause this to be parsed, then I don't think this will be necessary, as schema parsing is not currently part of performance benchmark

            validator.validate(URI.create(SpecificationVersion.DRAFT2019_09.getId()), "{}");
            return validator;
   default:
      throw new RuntimeException("Unsupported Spec:" + spec);
}
```

Reply:
Unfortunately meta-schemas will only get parsed when needed for the first time - with the first validation